csmaps_utils.sh

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env bash
+#
+# Copyright 2022 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -o nounset
+
+OC=oc
+YQ=yq
+CM_NAME="mock-cs-maps"
+CONTROL_NS="cs-control"
+
+# update_cs_maps Updates the common-service-maps with the given yaml. Note that
+# the given yaml should have the right indentation/padding, minimum 2 spaces per
+# line. If there are multiple lines in the yaml, ensure that each line has
+# correct indentation.
+function update_cs_maps() {
+    local yaml=$1
+
+    local object="$(
+        cat <<EOF
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: "$CM_NAME"
+  namespace: kube-public
+data:
+  common-service-maps.yaml: |
+${yaml}
+EOF
+)"
+    echo "$object" | oc apply -f -
+}
+
+# create_empty_csmaps Creates a new common-service-maps configmap and inserts
+# an empty common-service-maps.yaml field.
+#
+# If the common-service-maps already exists, then will error
+function create_empty_csmaps() {
+    title " Creating empty common-service-maps configmap "
+    local isExists=$("${OC}" get configmap --ignore-not-found -n kube-public "$CM_NAME")
+    if [ ! -z "$isExists" ]; then
+        info "The $CM_NAME already exists, skipping"
+        return
+    fi
+    update_cs_maps ""
+    success "Empty common-service-maps configmap created in kube-public namespace"
+}
+
+# insert_control_ns Insert the controlNamespace field into the configmap if it
+# does not exist
+function insert_control_ns() {
+    local current_yaml=$("${OC}" get -n kube-public cm "$CM_NAME" -o yaml | "${YQ}" '.data.["common-service-maps.yaml"]')
+
+    current=$(echo "$current_yaml" | "${YQ}" .'controlNamespace')
```

Comment:
does the period need to be inside the ''? '.controlNamespace' instead of .'controlNamespace'

Reply:
good catch. Weird that it still worked though